### PR TITLE
Bitcoin Blockchain Service

### PR DESCRIPTION
**Pull request by dinesh-skyach**
Added Bitcoin Blockchain Service

### CODE REVIEW

**Comment by ysv** [comment]
@dinesh-skyach can you share script which you used for bitcoin-data/1354419-1354420.json generations. Also can we somehow reduce the size of this file ?

**Comment by dinesh-skyach** [comment]
@ysv I did it manually via this
```
curl --data '{"jsonrpc":"2.0", "method":"getblock","params":["00000000bbc746ed23eef6d08c855b9ca07e64de84cb5d3846a667f60b0528cc", 2]}' http://user:user@127.0.0.1:18332

```

File: lib/client/bitcoin.rb [hunk 1]
```diff
@@ -62,15 +62,19 @@ def build_transaction(tx, current_block, latest_block, address)
 
         next if item.fetch('value').to_d <= 0
         next unless item['scriptPubKey'].has_key?('addresses')
-        next if address != item['scriptPubKey']['addresses'][0]
+        next if address != normalize_address(item['scriptPubKey']['addresses'][0])
```
Comment (ysv) [inline lib/client/bitcoin.rb]: @dinesh-skyach why do we use only the first address ?
`item['scriptPubKey']['addresses'][0]`

File: spec/services/blockchain_service/bitcoin_spec.rb [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        end
+      end
+
+      let(:currency) { Currency.find_by_id(:btc) }
+
+      before do
+        # Mock requests and methods.
+        client.class.any_instance.stubs(:latest_block_number).returns(latest_block)
+        client.class.any_instance.stubs(:get_block_hash).returns(block_data[0]["result"]["hash"], block_data[1]["result"]["hash"], block_data[2]["result"]["hash"])
+        client.class.any_instance.stubs(:get_block).returns(block_data[0]["result"], block_data[1]["result"], block_data[2]["result"])
```
Comment (ysv) [inline spec/services/blockchain_service/bitcoin_spec.rb]: Use stub_request here (example in ethereum_spec.rb)

**Review by ysv — CHANGES_REQUESTED** [review]

File: spec/services/blockchain_service/bitcoin_spec.rb (hunk 2 above)
Comment (dinesh-skyach) [reply spec/services/blockchain_service/bitcoin_spec.rb]: @ysv done

File: config/templates/seed/blockchains.yml.erb [hunk 3]
```diff
@@ -23,10 +23,10 @@
 - key:               btc-testnet
   name:              Bitcoin Testnet
   client:            bitcoin
-  server:            http://127.0.0.1:18332
-  height:            500000
+  server:            http://user:user@127.0.0.1:18332
```
Comment (ysv) [inline config/templates/seed/blockchains.yml.erb]: `json_rpc_endpoint: http://user:password@127.0.0.1:18332`

**Review by ysv — APPROVED** [review]

**Pull request merged**